project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc

```diff
@@ -62,20 +44,54 @@ function c4m_features_homepage_my_groups_content_type_render($subtype, $conf, $a
     )
   );
   $variables = array(
-    'first_groups' => $first_groups,
-    'extra_groups' => $extra_groups,
-    'show_all_link' => $count > 5 ? TRUE : FALSE,
+    'groups' => $rendered_groups,
     'link' => url('groups', $link_options),
   );
 
   $block = new stdClass();
   $block->module = 'c4m_features_homepage';
   $block->title = '';
-  $block->content = $first_groups ? theme('c4m_features_homepage_my_groups', $variables) : FALSE;
+  $block->content = $rendered_groups ? theme('c4m_features_homepage_my_groups', $variables) : FALSE;
   return $block;
 }
 
 /**
+ * Returns the list of groups of an user.
+ *
+ * @param object $account.
+ *   User object. Defaults global user if NULL.
+ *
+ * @returns mixed|null
+ *   List of groups or NULL if the user has no groups.
+ */
+function c4m_features_homepage_get_user_groups($account = NULL) {
+  $groups = c4m_user_og_get_groups($account);
+  usort($groups, 'c4m_sort_object_by_title');
+
+  return $groups;
+}
+
+/**
+ * Returns the list of groups of an user, grouped by the membership type.
+ *
+ * @param object $account.
+ *   User object. Defaults global user if NULL.
+ *
+ * @returns mixed|null
+ *   List of groups or NULL if the user has no groups.
+ */
+function c4m_features_homepage_get_user_groups_grouped_by_membership($account = NULL) {
+  $groups = c4m_features_homepage_get_user_groups($account, $allowed_types);
```

Comment:
@DavidHernandez 
we probably want to remove the `$allowed_types` from here too ...